apps/python/ruperts_challenge_app/telemetry_utility.py

```diff
@@ -18,22 +18,73 @@ def set_telemetry(self, telemetry):
             self.telemetry_access_lock.release()
 
     def get_telemetry_data(self, currentLaptime, currentSpline):
+
+        def weighted_interpolation(xs, ys, currentx):
```

Comment:
In my testing a more generalised approach enabling more values to be passed in seemed to help at the expense of the accuracy of the delta - I tried this with all weights 1 though. Need to just do some more testing/tuning. Need to write a program that learns the weights from you driving 😂 (not actually impossible to do as have access to the live actual delta...).

Reply:
I have a plan for how we could tune such a thing, or even tune these weightings better. If this proves not to be a good enough solution then I can figure out something smarter.